paimon-common/src/main/java/org/apache/paimon/types/DataField.java

```diff
@@ -76,6 +77,11 @@ public DataType type() {
         return type;
     }
 
+    @VisibleForTesting
```

Comment:
A newId method looks good to me.
We can just remove `VisibleForTesting`. (Public class should not have public VisibleForTesting method)